v2/options.go

```diff
@@ -26,6 +26,9 @@ type TopicPayload struct {
 	Payload string
 }
 
+// ConnectHandler is the function type for connect callback.
+type ConnectHandler func()
```

Comment:
I thing it's better to be able to receive the client as an argument
```suggestion
type ConnectHandler func(cli *DeviceClient)
```